packages/react/src/auto/hooks/useHasManyInputController.tsx

```diff
@@ -24,6 +26,13 @@ export const useHasManyInputController = (props: AutoRelationshipInputProps) =>
 
   const retrievedSelectedRecordIds = selected.records?.map((record: { id: string }) => record.id) ?? [];
   const unlinkedRecordIds = fields.filter((field: any) => field.__unlinkedInverseField).map((field: any) => field.__id);
+  const formContextValue = getValues(path);
+
+  useEffect(() => {
+    if (!formContextValue) {
+      void clearAllFields(); // This is called asynchronously to avoid an infinite loop
+    }
+  }, [!formContextValue || formContextValue.length === 0]);
```

Comment:
Why do we need to use `getValues` to see if it has value instead of the `fields` property from the `useFieldArray` hook?

Also I think the dependency array can be like this?
```
[clearAllFields, formContextValue]
```

Reply:
I use `getValues()` here because calling `useActionForm(...) .reset()` does not reset the values used in useFieldArray. They are only reset in `getValues`, thus leading to this synchronization process

And I made the dependency array like that so the useEffect would only run when emptied and unemptied. Having the whole `formContextValue` in there would cause it to run upon every change in the selection, even if the change would not be relevant to that process